docs/usage/types/dicts_mapping.md

```diff
@@ -18,3 +61,67 @@
     Prior to Python 3.8, it requires the [typing-extensions](https://pypi.org/project/typing-extensions/) package.
     But required and optional fields are properly differentiated only since Python 3.9.
     We therefore recommend using [typing-extensions](https://pypi.org/project/typing-extensions/) with Python 3.8 as well.
+
+Same as `dict` but Pydantic will validate the dictionary since keys are annotated.
+
+```py
+from typing import Optional
+
+from typing_extensions import TypedDict
+
+from pydantic import BaseModel, ConfigDict, ValidationError
+
+
+# `total=False` means keys are non-required
+class UserIdentity(TypedDict, total=False):
+    name: Optional[str]
+    surname: str
+
+
+class User(TypedDict):
+    __pydantic_config__ = ConfigDict(extra='forbid')  # type: ignore
```

Comment:
Can we link the docs where we talk about the `__pydantic_config__`? I think on `config_models.md`.

Also, how does the `type: ignore` looks in the documentation?

Reply:
> Can we link the docs where we talk about the __pydantic_config__? I think on config_models.md.

I've added a short description and a link to `config_models.md`

> Also, how does the type: ignore looks in the documentation?

It was ok but I've deleted it as it will pass without this as well.